### PR TITLE
Add option to mount CVMFS in skaha deployments

### PR DESCRIPTION
**Pull request by t1mk1k**
Add functionality to mount CVMFS in skaha deployments, by default this is turned off.

CVMFS will be mounted in:
- launch-carta.yaml
- launch-contributed.yaml
- launch-desktop-app.yaml
- launch-headless.yaml
- launch-notebook.yaml

and NOT in:
- launch-desktop.yaml

Changes:
- Introduce a variable to enable CVMFS (`deployment.skaha.cvmfsEnabled`)
- Add two helper functions for including the CVMFS Volume and VolumeMount in `deployment/helm/skaha/templates/_helpers.tpl`
- Adjust the control logic in `deployment/helm/skaha/templates/skaha-config-configmap.yaml` to:
  - use Helm templating logic on `deployment/helm/skaha/skaha-config/*.yaml` files
  - filter out none templating files
  - introduce scoping, so `.Files` and `.Values` are accessible in the range scope

### CODE REVIEW

**Review by at88mph — CHANGES_REQUESTED** [review]
Can you please also change the `README.md` in https://github.com/opencadc/science-platform/tree/master/deployment/helm?

Also update the `CHANGELOG.md` https://github.com/opencadc/science-platform/blob/master/deployment/helm/skaha/CHANGELOG.md.

**Comment by sfabbro** [comment]
May be worth repeating in the README.md the requirements for CVMFS on the hosts that are commented in the values.yml.

**Comment by t1mk1k** [comment]
> May be worth repeating in the README.md the requirements for CVMFS on the hosts that are commented in the values.yml.

@sfabbro: I've added a subsection for CVMFS to Skaha install: https://github.com/opencadc/science-platform/pull/645/files#
Is that what you were thinking of?

File: deployment/helm/skaha/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # CHANGELOG for Skaha User Session API (Chart 0.4.3)
 
+## 20224.06.28
```
Comment (at88mph) [inline deployment/helm/skaha/CHANGELOG.md]: Typo with `20224`

**Review by at88mph — CHANGES_REQUESTED** [review]

**Comment by at88mph** [comment]
Can something be clarified, please?  It's not obvious how the repository URL is specified if a Proxy is recommended.  Is the assumption that CVMFS is mounted on the Node(s), and the Pods will simply mount that from the Node's disk?

**Comment by rptaylor** [comment]
The proxy is external/unrelated to this. The URLs of stratum servers are based on CVMFS configuration which is also unrelated to this.

Yes this approach uses a CVMFS client installed and configured by cluster admins on the nodes.

File: deployment/helm/skaha/CHANGELOG.md (hunk 1 above)
Comment (t1mk1k) [reply deployment/helm/skaha/CHANGELOG.md]: Fixed

**Comment by t1mk1k** [comment]
> Can something be clarified, please? It's not obvious how the repository URL is specified if a Proxy is recommended. Is the assumption that CVMFS is mounted on the Node(s), and the Pods will simply mount that from the Node's disk?

As @rptaylor mentioned, the proxy setup is not relevant for the skaha deployment here but something that is part of the configuration of the CVMFS client that precedes the skaha installation/update.

I've removed the sentence about proxies from the README.md to avoid confusion and made it more explicit that the CVMFS client needs to be configured beforehand.

If it's not yet 100% clear, feel free to suggest a better sentence.

**Comment by rptaylor** [comment]
To be honest I'm starting to lean towards the CVMFS CSI approach instead of the hostPath approach. It would probably be more compatible/portable that way.

**Comment by brianmajor** [comment]
For info: there is some contention with [this PR](https://github.com/opencadc/science-platform/pull/681), and we will likely complete the CVMFS integration by iterating there.  Nothing required here now.

If we want to switch the the CSI approach let's do that later with another separate PR.

**Comment by brianmajor** [comment]
Closing this PR as it overlaps with the 'generic mounting' feature in #681  -  Thanks for your work on this @t1mk1k and reviews @rptaylor .  We plan on offering your CVMFS work (through the hostPath solution initially) to our user base.

**Pull request closed without merging**